src/wallet/wallet.h

```diff
@@ -200,12 +200,18 @@ class CNoteData
 public:
     libzcash::PaymentAddress address;
 
-    // It's okay to cache the nullifier in the wallet, because we are storing
-    // the spending key there too, which could be used to derive this.
-    // If PR #1210 is merged, we need to revisit the threat model and decide
-    // whether it is okay to store this unencrypted while the spending key is
-    // encrypted.
-    uint256 nullifier;
+    /**
+     * Cached note nullifier. May not be set if the wallet was not unlocked when
+     * this was CNoteData was created. If not set, we always assume that the
+     * note has not been spent.
+     *
+     * It's okay to cache the nullifier in the wallet, because we are storing
+     * the spending key there too, which could be used to derive this.
+     * If PR #1210 is merged, we need to revisit the threat model and decide
```

Comment:
Yes, it has. We never decided on whether this was an acceptable trade-off, but it is a necessary one in order to detect spends (and even then is not sufficient, as this PR shows).